packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,384 @@
+
+            self._channel._process_request("POST", url, param, body)
+
+    return HTTPHandler
+
+
+class HTTPConnection(Connection):
+    """Proxy to the functionality of the web RESTful API."""
+
+    def __init__(
+        self,
+        address: Address,
+        host: str = "",
+        port: int = 10000,
+        api_path: str = None,  # Directory path of the API YAML file.
+        *args,
+        **kwargs,
+    ):
+        """
+        Initialize a connection to an RESTful API.
+
+        :param api_spec_path: the directory path of the API specification YAML source file.
+        :param provider_addr: the provider IP address.
+        :param provider_port: the provider port.
+        :param connection_id: the identifier of the connection object.
+        :param restricted_to_protocols: the only supported protocols for this connection.
+        :param excluded_protocols: the excluded protocols for this connection.
+        """
+
+        # the following api_spec format checks will be in their own function check_api(api_spec)
+        if api_path is not None:
+            try:
+                with open(api_path, "r") as f:
+                    api_spec = yaml.safe_load(f)
+            except FileNotFoundError:
+                logger.error(
+                    "API specification YAML source file not found. Please double-check filename and path."
+                )
+                return
+        else:
+            api_spec = api_path
+
+        if kwargs.get("connection_id") is None:
+            kwargs["connection_id"] = PublicId("fetchai", "http", "0.1.0")
+
+        super().__init__(*args, **kwargs)
+        self.address = address
+        self.channel = HTTPChannel(
+            address,
+            host,
+            port,
+            api_spec,
+            connection_id=self.connection_id,
+            excluded_protocols=self.excluded_protocols,
+        )
+
+    async def connect(self) -> None:
+        """
+        Connect to the http.
+
+        :return: None
+        """
+        if not self.connection_status.is_connected:
+            self.connection_status.is_connected = True
+            self.channel.in_queue = asyncio.Queue()
+            self.channel.loop = self.loop
+            self.channel.connect()
+
+    async def disconnect(self) -> None:
+        """
+        Disconnect from HTTP.
+
+        :return: None
+        """
+        if self.connection_status.is_connected:
+            self.connection_status.is_connected = False
+            self.channel.disconnect()
+
+    async def send(self, envelope: "Envelope") -> None:
+        """
+        Send an envelope.
+
+        :param envelope: the envelop
+        :return: None
+        """
+        if not self.connection_status.is_connected:
+            raise ConnectionError(
+                "Connection not established yet. Please use 'connect()'."
+            )  # pragma: no cover
+        # self.channel.send(envelope)
```

Comment:
Note, this send function is called by the multiplexer to `send` outgoing envelopes via the connection.

Reply:
I have included send function in HTTPChannel.